From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Stephen Chenney <schenney@chromium.org>
Date: Fri, 24 Jul 2026 13:05:08 -0700
Subject: [HiC] Add privacy flag for text clip

Not really a bug since it's not an external resource, but it makes
the AI happy and prevents further bug reports or potential problems
if the code downstream ever paints cross origin content.

Bug: 523095011
Change-Id: Id6dea450cbea60a109dd67b1e2f53f8401e47015
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8139364
Commit-Queue: Philip Rogers <pdr@chromium.org>
Auto-Submit: Stephen Chenney <schenney@chromium.org>
Reviewed-by: Philip Rogers <pdr@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1668117}

diff --git a/third_party/blink/renderer/core/paint/box_fragment_painter.cc b/third_party/blink/renderer/core/paint/box_fragment_painter.cc
index d5045b1e2111b55499a1597b1d7f4a912fa1f4da..c2d0486924570eb1f02f05af21048dcc67c3b2c9 100644
--- a/third_party/blink/renderer/core/paint/box_fragment_painter.cc
+++ b/third_party/blink/renderer/core/paint/box_fragment_painter.cc
@@ -2282,9 +2282,11 @@ void BoxFragmentPainter::PaintTextClipMask(const PaintInfo& paint_info,
                                            const gfx::Rect& mask_rect,
                                            const PhysicalOffset& paint_offset,
                                            bool object_has_multiple_boxes) {
-  PaintInfo mask_paint_info(paint_info.context, CullRect(mask_rect),
-                            PaintPhase::kTextClip,
-                            paint_info.DescendantPaintingBlocked());
+  PaintInfo mask_paint_info(
+      paint_info.context, CullRect(mask_rect), PaintPhase::kTextClip,
+      paint_info.DescendantPaintingBlocked(),
+      paint_info.IsPrivacyPreserving() ? PaintFlag::kPrivacyPreserving
+                                       : PaintFlag::kNoFlag);
   if (!object_has_multiple_boxes) {
     PaintObject(mask_paint_info, paint_offset);
     return;
